libretro-common/include/libretro.h

```diff
@@ -976,6 +976,32 @@ struct retro_hw_render_context_negotiation_interface
                                             * so it will be used after SET_HW_RENDER, but before the context_reset callback.
                                             */
 
+#define RETRO_SERIALIZATION_QUIRK_INCOMPLETE (1 << 0)
+   /* Serialized state is incomplete in some way. Set if serialization is
+    * usable in typical end-user cases but should not be relied upon to
+    * implement frame-sensitive frontend features such as netplay or
+    * rerecording. */
+#define RETRO_SERIALIZATION_QUIRK_MUST_INITIALIZE (1 << 1)
+   /* The core must spend some time initializing before serialization is
+    * safe. */
+#define RETRO_SERIALIZATION_QUIRK_INITIALIZING (1 << 2)
+   /* If MUST_INITIALIZE is set, this should also be set if initialization is
+    * in progress. */
```

Comment:
I guess I misunderstood or didn't properly communicate. That solution is more elegant, but the current solution is easier to mash into existing cores. More crucially, serialize_size may also fail before initialization, and it has no way to communicate failure except to return size 0. Perhaps that's sufficient?